crates/core/src/db/datastore/locking_tx_datastore/mod.rs

```diff
@@ -461,10 +463,9 @@ impl Inner {
             *sequences_start.entry(ST_INDEXES_ID).or_default() += 1;
         }
 
-        // TODO(kim): We need to make sure to have ST_MODULE in the committed
-        // state. `bootstrap_system_table` initializes the others lazily, but
-        // it doesn't know about `ST_MODULE_ROW_TYPE`. Perhaps the committed
-        // state should be initialized eagerly here?
+        // We don't add the row here but with `MutProgrammable::set_program_hash`, but we need to register the table
+        // in the internal state.
+        // TODO: This should be moved here?
```

Comment:
You mean set_program_hash should be moved here? That would only be possible if we could bootstrap in a transaction -- otherwise, if the init reducer fails, the hash may point to nothing.

Reply:
Understood, I remove the TODO then.